types/params.go

```diff
@@ -141,10 +154,27 @@ type SynchronyParams struct {
 // The goal is facilitate the progression of consensus when improper synchrony
 // parameters are set or become insufficient to preserve liveness. Refer to
 // https://github.com/cometbft/cometbft/issues/2184 for more details.
+//
+// There's a cap (MaxMessageDelay) on the MessageDelay to prevent overflow.
 func (sp SynchronyParams) InRound(round int32) SynchronyParams {
+	if round <= 0 {
+		return sp
+	}
+
+	d := time.Duration(math.Pow(1.1, float64(round)) * float64(sp.MessageDelay))
```

Comment:
This here still can cause an overflow if `round` is large enough. You need to check the duration before you cast it to an int64, i.e., time.Duration.

Again, we should not overflow, and then try and fix it. We should just prevent the overflows.

If you do this it will be safe and not overflow, you can also remove the if statement,

```
d := time.Duration(math.Min(
  float64(MaxMessageDelay),
  math.Pow(1.1, float64(round)) * float64(sp.MessageDelay),
))
```

Reply:
[`9183157` (#4816)](https://github.com/cometbft/cometbft/pull/4816/commits/918315707ed9185b372a3cab3375b31935bd50d1)